packages/app/src/components/nominationInfo/ApplicationForm.js

```diff
@@ -0,0 +1,311 @@
+  }, [props.formData['Discharge Date']]);
+
+  const phoneRegex = /^[\+]?[(]?[0-9]{3}[)]?[-\s\.]?[0-9]{3}[-\s\.]?[0-9]{4,6}$/;
+  const yesNoRegex = /^(?:Yes\b|No\b)/;
+
+  useEffect(() => {
+    if (!firstUpdate.current) {
+      handleSubmit(submitForm)();
+    }
+    firstUpdate.current = false;
+  }, [props.saveHasBeenClicked]);
+
+  const validationSchema = Yup.object({
+    'Admission Date': Yup.date().required('Required'),
+    'Discharge Date': Yup.date()
+      .min(
+        Yup.ref('Admission Date'),
+        'Discharge date cannot be before admission date.'
+      )
+      .required('Required'),
+    'Representative Name': Yup.string()
+      .min(3, 'Must be 3 characters or more.')
+      .max(30, 'Must be 30 characters or less.')
+      .required('Required'),
+    'Representative Email Address': Yup.string()
+      .email('Invalid email address.')
+      .required('Required'), // This handles email validation with no regex.
+    'Representative Phone Number': Yup.string()
+      .matches(phoneRegex, 'Please enter a valid phone number.')
+      .required('Required'),
+    Relationship: Yup.string()
+      .min(3, 'Must be at least 3 characters.')
+      .max(20, 'Must be no more than 20 characters.')
+      .required('Required'),
+    'Request to communicate in Spanish?': Yup.string()
+      .matches(yesNoRegex, 'Please enter yes or no.')
+      .required('Required'),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm({
+    resolver: yupResolver(validationSchema),
+  });
+
+  const submitForm = async data => {
+    if (NominationsData) {
+      let newActiveNomination = {};
+      const newNominationData = NominationsData.map(nomination => {
+        if (nomination.id === props.id) {
+          data['Admission Date']
+            ? (nomination.admissionDate = data[
+                'Admission Date'
+              ].toLocaleDateString())
+            : (nomination.admissionDate = data['Admission Date']);
+          data['Discharge Date']
+            ? (nomination.dischargeDate = data[
+                'Discharge Date'
+              ].toLocaleDateString())
+            : (nomination.dischargeDate = data['Discharge Date']);
+          nomination.representativeEmailAddress =
+            data['Representative Email Address'];
+          nomination.representativePhoneNumber =
+            data['Representative Phone Number'];
+          nomination.representativeRelationship = data['Relationship'];
+          nomination.representativeName = data['Representative Name'];
+          if (data['Request to communicate in Spanish?'] === 'Yes') {
+            nomination.representativeSpanishRequested = true;
+          } else {
+            nomination.representativeSpanishRequested = false;
+          }
+          newActiveNomination = nomination;
+        }
+        return nomination;
+      });
+      const requestBody = {
+        admissionDate: newActiveNomination.admissionDate,
```

Comment:
Could the request body just be the `newActiveNomination` object instead of mapping to a new object here?

Reply:
yes @t-keazirian let's send the whole nomination object to the backend.